src/rgw/rgw_lc.cc

```diff
@@ -1923,13 +1970,102 @@ time_t RGWLC::thread_stop_at()
   return time(nullptr) + interval;
 }
 
+int RGWLC::process_bucket(int index, int max_lock_secs, LCWorker* worker,
+			  const std::string& bucket_entry_marker,
+			  bool once = false)
+{
+  ldpp_dout(this, 5) << "RGWLC::process_bucket(): ENTER: "
+	  << "index: " << index << " worker ix: " << worker->ix
+	  << dendl;
+
+  int ret = 0;
+  rgw::sal::LCSerializer* lock = sal_lc->get_serializer(lc_index_lock_name,
+							obj_names[index],
+							std::string());
+  auto lock_rele = make_scope_guard(
+    [&lock] {
+      delete lock;
+    }
+    );
```

Comment:
i think you may have misread my comment suggesting `std::scoped_lock` to handle the unlock with RAII

to handle the deletion of this memory, you can just use `std::unique_ptr<rgw::sal::LCSerializer> lock`

(though my comment about `std::scoped_lock` missed the fact that you need to manually unlock before calling `bucket_lc_process()`, so you'd need to use `std::unique_lock` there instead)

Reply:
I did not misread your comment.  Just to be clear, are you saying that you object to my change if it does not adopt your exact preferences regarding smart pointers--and how much other code to change?

Reply:
that comment was about replacing gotos with raii, which you seemed to agree with, but i still see gotos and manual calls to `unlock()`

Reply:
ok, I won't try to argue in favor of gotos here :)